portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,62 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.comment.CommentStorage;
+import com.google.sps.config.Constants;
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles adding and fetching comments in individual sections
+ *  The server also handles the html format of the comments.
+ */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private CommentStorage commentStorage = new CommentStorage();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Step 1 done!</h1>");
+
+    response.getWriter().println(
+             formatCommentsBulk(commentStorage.
+             getStorageEntries("Comment", request.getQueryString())));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    commentStorage.addStorageEntry(request.getParameter("username"),
+                                   request.getParameter("comment"),
+                                   request.getParameter("image-id"),
+                                   "Comment");
+  }
+
+  private String formatCommentsBulk(final Iterable<Entity> comments) {
```

Comment:
Consider adding the `@Nullable` annotation to the `comments` argument to make it clear that this method works when that is `null`.